File: ncms-engine/ncms-engine-core/src/main/java/com/softmotions/ncms/marketing/mtt/MttRuleFilter.java
package com.softmotions.ncms.marketing.mtt;

import java.io.Serializable;
import java.util.Date;
import java.util.Objects;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * @author Tyutyunkov Vyacheslav (tve@softmotions.com)
 */
@XmlAccessorType(XmlAccessType.NONE)
@JsonAutoDetect(
        creatorVisibility = JsonAutoDetect.Visibility.NONE,
        fieldVisibility = JsonAutoDetect.Visibility.NONE,
        getterVisibility = JsonAutoDetect.Visibility.NONE,
        isGetterVisibility = JsonAutoDetect.Visibility.NONE,
        setterVisibility = JsonAutoDetect.Visibility.NONE
)
public class MttRuleFilter implements Serializable {
    @JsonProperty
    private Long id;
    @JsonProperty
    private Long ruleId;
    @JsonProperty
    private String type;
    @JsonProperty
    private String description;
    @JsonProperty
    private String spec;
    @JsonProperty
    private Date cdate;
    @JsonProperty
    private Date mdate;
    @JsonProperty
    private boolean enabled = true;

    public MttRuleFilter() {
    }

    public MttRuleFilter(Long ruleId, String type) {
        this.ruleId = ruleId;
        this.type = type;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getRuleId() {
        return ruleId;
    }

    public void setRuleId(Long ruleId) {
        this.ruleId = ruleId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getSpec() {
        return spec;
    }

    public void setSpec(String spec) {
        this.spec = spec;
    }

    public Date getCdate() {
        return cdate;
    }

    public void setCdate(Date cdate) {
        this.cdate = cdate;
    }

    public Date getMdate() {
        return mdate;
    }

    public void setMdate(Date mdate) {
        this.mdate = mdate;
    }

    public boolean getEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MttRuleFilter that = (MttRuleFilter) o;
        return Objects.equals(id, that.id);
    }

    public int hashCode() {
        return Objects.hash(id);
    }
}

File: ncms-engine/ncms-engine-core/src/main/java/com/softmotions/ncms/marketing/mtt/MttRuleFilter.kt
package com.softmotions.ncms.marketing.mtt

import java.io.Serializable
import java.util.*

/**
 * @author Tyutyunkov Vyacheslav (tve@softmotions.com)
 */
data class MttRuleFilter(var id: Long = 0,
                         var ruleId: Long? = null,
                         var type: String? = null,
                         var description: String? = null,
                         var spec: String? = null,
                         var cdate: Date? = null,
                         var mdate: Date? = null,
                         var enabled: Boolean? = true

) : Serializable {

    @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
    constructor(id: java.lang.Long,
                ruleId: java.lang.Long,
                type: java.lang.String,
                description: java.lang.String,
                spec: java.lang.String,
                cdate: java.sql.Timestamp,
                mdate: java.sql.Timestamp,
                enabled: java.lang.Boolean)
    : this(id.toLong(), ruleId.toLong(), type.toString(), description.toString(),
            spec.toString(), Date(cdate.time), Date(mdate.time),
            enabled.booleanValue())
}
